rscripts/YAL5-codon-pairs.R

```diff
@@ -0,0 +1,1455 @@
+
+  # select the filtered counts (length(filtered_counts) == length(cds_length))
+  filtered_counts <- filtered$Count
+
+  return(filtered_counts)
+}
+# TEST: FilterForFrame(): returns a list of numeric values = TRUE
+# TEST: FilterForFrame(): length(filtered_counts) = length(cds_length) for gene
+#                         of interest = TRUE
+# gives:
+# > str(filtered_counts)
+# num [1:207] 811 429 488 102 994 146 173 762 13 176 ...
+
+
+### Fetch and format counts from the h5 file ###
+
+
+#' GetAllCodonPosCounts(): extracts A-site assigned counts for a list of genes
+#'
+#' This function extracts the A-site assigned counts and generates a tidy data
+#' frame (tibble) which contains the counts for all genes in the list of genes.
+#'
+#' If filter_for_frame = FALSE the GetGeneCodonPosReads1dsnap() function is
+#' applied to a list of genes and generates a tibble which contains the counts
+#' from all reading frames.
+#'
+#' If filter_for_frame = TRUE the FilterForFrame() function is applied instead,
+#' this generates a tibble which contains the read counts for a reading frame
+#' (snapdisp = 0L or 1L or 2L).
+#'
+#' @param gene from gene_names to get read lengths for
+#' @param dataset name of dataset stored in .h5 file.
+#' @param hd_file name of .h5 hdf5 file holding read data for all genes,
+#' created from BAM files for dataset samples.
+#' @param min_read_length integer, minimum read length in H5 output;
+#' Default = 10 (set in generate_stats_figs.R from yaml)
+#' @param snapdisp integer any additional displacement in the snapping;
+#' Default = 0L
+#' @param asite_disp_path integer, lengths used for a-site assignment
+#' @param filter_for_frame TRUE if filtering for a reading frame, FALSE if
+#' keeping and grouping all reading frames for each codon
+#'
+#' @return a tibble which contains the columns "Gene", "PosCodon" and "Count"
+#' for a list of genes
+#'
+#' @example
+#'
+#' GetAllCodonPosCounts(gene_names,
+#'                      dataset = "Mok-simYAL5",
+#'                      hd_file = here::here("Mok-simYAL5",
+#'                                           "output",
+#'                                           "A",
+#'                                           "A.h5"),
+#'                      min_read_length = 10,
+#'                      snapdisp = 0L,
+#'                      asite_disp_path = here::here(
+#'                        "data",
+#'                        "yeast_standard_asite_disp_length.txt"),
+#'                      filter_for_frame = TRUE)
+#'
+#' @export
+GetAllCodonPosCounts <- function(gene_names, dataset, hd_file, min_read_length,
+                                 snapdisp, asite_disp_path,
+                                 filter_for_frame) {
+
+  # fetch the gene names contained within the h5 file
+  gene_names <- rhdf5::h5ls(hd_file, recursive = 1)$name
+
+  # fetch values used for asite assignment
+  asite_disp_length <- ReadAsiteDisplacementLengthFromFile(asite_disp_path)
```

Comment:
`ReadAsiteDisplacementLengthFromFile(asite_disp_path)` -> `suppressMessages(ReadAsiteDisplacementLengthFromFile(asite_disp_path))`

This will prevent 
```
-- Column specification ----------------------------------------------------------------------------------------------------------------------------------------------
cols(
  read_length = col_double(),
  asite_displacement = col_double()
)
```
from being printed to the console every time `GetAllCodonPosCounts` is called